server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -197,7 +199,14 @@ public void writeTo(StreamOutput out) throws IOException {
         if (hasFallback) {
             new WriteableSetting(setting.fallbackSetting, type).writeTo(out);
         }
-        // We are using known types so don't need the parser
+        // Write a boolean specifying whether the parser is an instanceof writeable
+        boolean isParserWriteable = parser instanceof Writeable;
+        if (isParserWriteable = true) {
+            out.writeBoolean(isParserWriteable);
```

Comment:
You're writing this in both the `if` and `else` cases, so you might just want to write it before the conditional (and then you wouldn't need an `else`).  SEe how `hasFallback` was done a few lines up.